addons/sale/models/sale.py

```diff
@@ -838,7 +838,8 @@ def _compute_tax_id(self):
         for line in self:
             fpos = line.order_id.fiscal_position_id or line.order_id.partner_id.property_account_position_id
             # If company_id is set, always filter taxes by the company
-            taxes = line.product_id.taxes_id.filtered(lambda r: not line.company_id or r.company_id == line.company_id)
+            line_company_id = line.company_id or line.order_id.company_id
```

Comment:
As line.company_id is a related to the order's company, is it usefull ?

Reply:
well, this is the issue here.
when passing here as someone else than the admin, line.company_id is not set
but since the flow implies that we pass order_id in the vals, the "new" record in the onchange has that value